pkg/test/e2e/konnectivity/konnectivity_test.go

```diff
@@ -264,13 +264,49 @@ func TestKonnectivity(t *testing.T) {
 
 	t.Log("check if konnectivity-agents are deployed")
 	{
-		pods, err := getPods(ctx, userClient, "konnectivity-agent")
-		if err != nil {
-			t.Errorf("failed to get konnectivity-agent pods: %s", err)
-		}
+		err := wait.Poll(30*time.Second, 10*time.Minute, func() (bool, error) {
+			pods, err := getPods(ctx, userClient, "konnectivity-agent")
+			if err != nil {
+				t.Logf("failed to get konnectivity-agent pods: %s", err)
+				return false, nil
+			}
+
+			if len(pods) != 2 {
+				t.Logf("expected 2 konnectivity-agent pods got: %d", len(pods))
+				return false, nil
+			}
+
+			// check if they are running and healthy
+			allHealthy := true
+			for _, pod := range pods {
+				if pod.Status.Phase != corev1.PodRunning {
+					allHealthy = false
+				}
+				for _, c := range pod.Status.Conditions {
+					if c.Type == corev1.PodReady {
+						if c.Status != corev1.ConditionTrue {
+							allHealthy = false
+							t.Log("not ready", pod.Name, c.Type, c.Status)
+						}
+					} else if c.Type == corev1.ContainersReady {
+						if c.Status != corev1.ConditionTrue {
+							allHealthy = false
+							t.Log("not container ready", pod.Name, c.Type, c.Status)
+						}
+					}
+				}
+				t.Log(pod.Name, pod.Status.Phase)
```

Comment:
Here you also have a similar code block as in the cilium tests: get a list of pods, and check if they are ready, you can deduplicate all of this if you extract that to a func.